src/SonarQube.Client/Services/SonarQubeClient.cs

```diff
@@ -0,0 +1,282 @@
+                async (HttpResponseMessage response) =>
+                {
+                    var byteArray = await response.Content.ReadAsByteArrayAsync();
+                    // Protobuf for C# throws when trying to read outside of the buffer and ReadAsStreamAsync returns a non
+                    // seekable stream so we can't determine when to stop. The hack is to use an intermediate MemoryStream
+                    // so we can control when to stop reading.
+                    // Note we might want to use FileStream instead to avoid intensive memory usage.
+                    using (var stream = new MemoryStream(byteArray))
+                    {
+                        return ReadFromProtobufStream(stream, ServerIssue.Parser).ToArray();
+                    }
+                });
+        }
+
+        public Task<Result<OrganizationResponse[]>> GetOrganizationsAsync(OrganizationRequest request, CancellationToken token)
+        {
+            const string OrganizationsAPI = "api/organizations/search"; // Since 6.2; internal
+
+            var query = AppendQueryString(OrganizationsAPI, "?p={0}&ps={1}", request.Page, request.PageSize); // TODO: should handle optional params
+
+            return InvokeSonarQubeApi(query, token,
+                stringResponse => JObject.Parse(stringResponse)["organizations"].ToObject<OrganizationResponse[]>());
+        }
+
+        public Task<Result<PluginResponse[]>> GetPluginsAsync(CancellationToken token)
+        {
+            const string ServerPluginsInstalledAPI = "api/updatecenter/installed_plugins"; // Since 2.10; internal
+
+            return InvokeSonarQubeApi(ServerPluginsInstalledAPI, token,
+                stringResponse => ProcessJsonResponse<PluginResponse[]>(stringResponse, token));
+        }
+
+        public Task<Result<ProjectResponse[]>> GetProjectsAsync(CancellationToken token)
+        {
+            const string ProjectsAPI = "api/projects/index"; // Since 2.10
+
+            return InvokeSonarQubeApi(ProjectsAPI, token,
+                stringResponse => ProcessJsonResponse<ProjectResponse[]>(stringResponse, token));
+        }
+
+        public Task<Result<PropertyResponse[]>> GetPropertiesAsync(CancellationToken token)
+        {
+            const string PropertiesAPI = "api/properties/"; // Since 2.6
+
+            return InvokeSonarQubeApi(PropertiesAPI, token,
+                stringResponse => ProcessJsonResponse<PropertyResponse[]>(stringResponse, token));
+        }
+
+        public Task<Result<QualityProfileChangeLogResponse>> GetQualityProfileChangeLogAsync(
+            QualityProfileChangeLogRequest request, CancellationToken token)
+        {
+            const string QualityProfileChangeLogAPI = "api/qualityprofiles/changelog"; // Since 5.2
+
+            var query = AppendQueryString(QualityProfileChangeLogAPI, "?profileKey={0}&ps={1}",
+                        request.QualityProfileKey, request.PageSize); // TODO: should handle optional params
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this 'TODO' comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=csharpsquid%3AS1135)